server/datastore/mysql/hosts.go

```diff
@@ -58,7 +58,8 @@ func (d *Datastore) NewHost(host *fleet.Host) (*fleet.Host, error) {
 }
 
 func (d *Datastore) SaveHost(host *fleet.Host) error {
-	sqlStatement := `
+	return d.withRetryTxx(func(tx *sqlx.Tx) error {
```

Comment:
This is one where _technically_, I agree completely. But _practically_, I wonder if we wouldn't be better off doing it without a tx, so that if, say, software fails later on, but pack stats don't so at least you see that part.

Reply:
Totally agreed. @zwass any reason why not go in this direction?

Reply:
I am on board with "succeed or fail as a whole".